src/Raven.Client/Documents/Session/DocumentSession.Attachments.cs

```diff
@@ -19,24 +21,44 @@ namespace Raven.Client.Documents.Session
     /// </summary>
     public partial class DocumentSession
     {
-        public AttachmentResult GetAttachment(string documentId, string name, Action<AttachmentResult, Stream> stream)
+        public AttachmentName[] GetAttachmentNames(object entity)
         {
-            var operation = new GetAttachmentOperation(documentId, name, stream, AttachmentType.Document, null);
+            if (entity == null)
+                return null;
+
+            if (DocumentsByEntity.TryGetValue(entity, out DocumentInfo document) == false)
+                ThrowEntityNotInSession(entity);
+
+            if (document.Metadata.TryGet(Constants.Documents.Metadata.Attachments, out BlittableJsonReaderArray attachments) == false)
+                return null;
+
+            var results = new AttachmentResult[attachments.Length];
+            for (var i = 0; i < attachments.Length; i++)
+            {
+                var attachment = (BlittableJsonReaderObject)attachments[i];
+                results[i] = JsonDeserializationClient.AttachmentResult(attachment);
+            }
+            return results;
+        }
+
+        public AttachmentResultWithStream GetAttachment(string documentId, string name)
```

Comment:
Change it so it has two overloads on `Stream GetAttachment(...)` the second `Stream GetAttachment(..., out AttchmentDetails d`)